server/src/main/java/org/elasticsearch/health/HealthIndicatorResult.java

```diff
@@ -17,6 +17,10 @@ public record HealthIndicatorResult(String name, String component, HealthStatus
     implements
         ToXContentObject {
 
+    public static HealthIndicatorResult of(String name, String component, HealthStatus status, String summary) {
+        return new HealthIndicatorResult(name, component, status, summary, (builder, params) -> builder);
```

Comment:
Nit: consider using `HealthIndicatorDetails.EMPTY` instead of `(builder, params) -> builder`

Reply:
Thanks for the hint!